### ISSUE TITLE
The new --backup-dir option does not work with OneDrive

### ISSUE THREAD

**Issue by traynier** [ncw/rclone#1224]
I originally added a comment to #98 but as that issue is closed you might not see it, so adding as a new issue just in case.

> What is your rclone version (eg output from `rclone -V`)
rclone v1.35-163-gc45c604β

> Which OS you are using and how many bits (eg Windows 7, 64 bit)
Windows 10 64 bit

> Which cloud storage system are you using? (eg Google Drive)
OneDrive.  I originally found this using crypt and encrypted filenames, but the same thing happens using a base OneDrive (un-encrypted) remote.

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

Note test is a local folder, with ./test/test.txt file in it, edited so it differs to the first time rclone was used.

rclone sync test one:Z/test --backup-dir one:Zold -vv

> A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

When one:Zold directory does NOT exist:

2017/03/08 19:12:40 DEBUG : rclone: Version "v1.35-163-gc45c604β" starting with parameters ["rclone" "sync" "test" "one:Z/test" "--backup-dir" "one:Zold" "-vv"]
2017/03/08 19:12:41 INFO  : One drive root 'Z/test': Modify window is 1s
2017/03/08 19:12:41 DEBUG : One drive root 'Z/test': Reading ""
2017/03/08 19:12:41 DEBUG : One drive root 'Z/test': Finished reading ""
2017/03/08 19:12:41 INFO  : One drive root 'Z/test': Waiting for checks to finish
2017/03/08 19:12:41 DEBUG : test.txt: Sizes differ
2017/03/08 19:12:41 ERROR : test.txt: Failed to copy: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:41 ERROR : test.txt: Not deleting source as copy failed: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:41 INFO  : One drive root 'Z/test': Waiting for transfers to finish
2017/03/08 19:12:41 ERROR : One drive root 'Z/test': not deleting files as there were IO errors
2017/03/08 19:12:41 ERROR : Attempt 1/3 failed with 1 errors and: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:42 DEBUG : One drive root 'Z/test': Reading ""
2017/03/08 19:12:42 DEBUG : One drive root 'Z/test': Finished reading ""
2017/03/08 19:12:42 DEBUG : test.txt: Sizes differ
2017/03/08 19:12:42 INFO  : One drive root 'Z/test': Waiting for checks to finish
2017/03/08 19:12:42 ERROR : test.txt: Failed to copy: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:42 ERROR : test.txt: Not deleting source as copy failed: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:42 INFO  : One drive root 'Z/test': Waiting for transfers to finish
2017/03/08 19:12:42 ERROR : One drive root 'Z/test': not deleting files as there were IO errors
2017/03/08 19:12:42 ERROR : Attempt 2/3 failed with 1 errors and: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:43 DEBUG : One drive root 'Z/test': Reading ""
2017/03/08 19:12:43 DEBUG : One drive root 'Z/test': Finished reading ""
2017/03/08 19:12:43 DEBUG : test.txt: Sizes differ
2017/03/08 19:12:43 INFO  : One drive root 'Z/test': Waiting for checks to finish
2017/03/08 19:12:43 ERROR : test.txt: Failed to copy: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:43 ERROR : test.txt: Not deleting source as copy failed: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:43 INFO  : One drive root 'Z/test': Waiting for transfers to finish
2017/03/08 19:12:43 ERROR : One drive root 'Z/test': not deleting files as there were IO errors
2017/03/08 19:12:43 ERROR : Attempt 3/3 failed with 1 errors and: can't copy "test.txt" -> "test.txt" as are same name when lowercase
2017/03/08 19:12:43 Failed to sync: can't copy "test.txt" -> "test.txt" as are same name when lowercase

I then manually created the one:Zold folder (using the impressive Rclone Browser app), but get the same error on running rclone again.

But calling rclone with --suffix works:
    rclone sync test one:Z/test --backup-dir one:Zold --suffix ".old" -vv

But it only works once.  Editing the test/test.txt file, and calling it again (i.e. when one:Zold/test.txt.old already exists from the previous time)

2017/03/08 19:15:25 DEBUG : rclone: Version "v1.35-163-gc45c604β" starting with parameters ["rclone" "sync" "test" "one:Z/test" "--backup-dir" "one:Zold" "--suffix" ".old" "-vv"]
2017/03/08 19:15:26 INFO  : One drive root 'Z/test': Modify window is 1s
2017/03/08 19:15:26 DEBUG : One drive root 'Z/test': Reading ""
2017/03/08 19:15:27 DEBUG : One drive root 'Z/test': Finished reading ""
2017/03/08 19:15:27 DEBUG : test.txt: Sizes differ
2017/03/08 19:15:27 INFO  : One drive root 'Z/test': Waiting for checks to finish
2017/03/08 19:15:28 DEBUG : pacer: Rate limited, increasing sleep to 20ms
2017/03/08 19:15:28 DEBUG : pacer: low level retry 1/10 (error 500 Internal Server Error: )
2017/03/08 19:15:29 DEBUG : pacer: Rate limited, increasing sleep to 40ms
2017/03/08 19:15:29 DEBUG : pacer: low level retry 2/10 (error 500 Internal Server Error: )
2017/03/08 19:15:29 DEBUG : pacer: Rate limited, increasing sleep to 80ms
2017/03/08 19:15:29 DEBUG : pacer: low level retry 3/10 (error 500 Internal Server Error: )
2017/03/08 19:15:29 DEBUG : pacer: Rate limited, increasing sleep to 160ms
2017/03/08 19:15:29 DEBUG : pacer: low level retry 4/10 (error 500 Internal Server Error: )
2017/03/08 19:15:29 DEBUG : pacer: Rate limited, increasing sleep to 320ms
2017/03/08 19:15:29 DEBUG : pacer: low level retry 5/10 (error 500 Internal Server Error: )
2017/03/08 19:15:29 DEBUG : pacer: Rate limited, increasing sleep to 640ms
2017/03/08 19:15:29 DEBUG : pacer: low level retry 6/10 (error 500 Internal Server Error: )
2017/03/08 19:15:30 DEBUG : pacer: Rate limited, increasing sleep to 1.28s
2017/03/08 19:15:30 DEBUG : pacer: low level retry 7/10 (error 500 Internal Server Error: )
2017/03/08 19:15:30 DEBUG : pacer: Rate limited, increasing sleep to 2s
2017/03/08 19:15:30 DEBUG : pacer: low level retry 8/10 (error 500 Internal Server Error: )
2017/03/08 19:15:31 DEBUG : pacer: low level retry 9/10 (error 500 Internal Server Error: )
2017/03/08 19:15:33 DEBUG : pacer: low level retry 10/10 (error 500 Internal Server Error: )
2017/03/08 19:15:33 DEBUG : test.txt: Received error: 500 Internal Server Error:  - low level retry 1/10
2017/03/08 19:15:35 DEBUG : pacer: Reducing sleep to 1.5s
2017/03/08 19:15:37 DEBUG : pacer: Rate limited, increasing sleep to 2s
2017/03/08 19:15:37 DEBUG : pacer: low level retry 1/10 (error 500 Internal Server Error: )

(this is the repeated call to monitor I mentioned in my comment on #98)

**Comment by ncw** [comment]
The integration tests I run before a release have pointed this up too!

I see what you mean about the monitor returning 500 with a valid JSON blob.  That is unexpected!  Will have to read the API docs again.

**Comment by ncw** [comment]
I've fixed this here

http://beta.rclone.org/v1.35-165-gf046c00/ (uploaded in 15-30 mins)

Let me know if it works for you!

**Comment by ncw** [comment]
I fixed the parsing of the async status too in

http://beta.rclone.org/v1.35-167-g0faf827/ (uploaded in 15-30 mins)

**Comment by traynier** [comment]
Great - yes, I no longer get the errors or loop endlessly running monitor!  👍 


However, I still get an unexpected result when the backup-dir does not exist on the destination.

i.e. do the following:

; create a file in the test folder and copy folder to onedrive
mkdir test
echo 1 >> test/test.txt
rclone copy ./test one:test --backup-dir one:testold -vv

; change the file and run the copy again
echo 2 >> test/test.txt
rclone copy ./test one:test --backup-dir one:testold -vv


At this point I would expect one:testold/test.txt to contain the old version.  But it does not use the backup-dir, and instead the old test.txt ends up in the root folder as one:test.txt

If the one:testold directory exists then it does work as expected.  However I had to manually create it (I used rclone browser).

Continuing from that, if I make another change

; change the file and run the copy again
echo 3 >> test/test.txt
rclone copy ./test one:test --backup-dir one:testold -vv

it then fails:

  Failed to copy: test.txt: async operation "ItemCopy" returned "failed"

presumably due to the file one:testold/test.txt already existing?




Repeating the test for an encrypted onedrive remote get slightly different result; if the backup-dir does not exist you just get an error instead of the old file ending up in the root folder:

  Failed to copy: 002i3q2u20jlcd0ietvoa93ri8: async operation "ItemCopy" returned "failed"

But otherwise works the same (works if no file in testold, fails if file already in testold).
